bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/selector/TimeRangeSelector.scala

```diff
@@ -1,25 +1,28 @@
 package edu.gemini.qv.plugin.selector
 
-import edu.gemini.qv.plugin.{TimeZoneChanged, TimeValueChanged, TimeRangeChanged, QvContext}
+import edu.gemini.qv.plugin.{QvContext, TimeRangeChanged, TimeValueChanged, TimeZoneChanged}
 import edu.gemini.qv.plugin.selector.TimeRangeSelector._
-import edu.gemini.qv.plugin.ui.{QvGui, CalendarDialog}
+import edu.gemini.qv.plugin.ui.{CalendarDialog, QvGui}
 import edu.gemini.shared.gui.monthview.DateSelectionMode
-import edu.gemini.skycalc.TimeUtils
 import edu.gemini.spModel.core.Semester
 import java.awt.event.{AdjustmentEvent, AdjustmentListener}
 import java.util.TimeZone
+
+import edu.gemini.shared.util.DateTimeUtils
+
+import scala.concurrent.duration._
 import scala.swing._
 import scala.swing.event.ButtonClicked
 
 object TimeRangeSelector {
   abstract class RangeType(val duration: Long, val increment: Long)
-  object Day extends RangeType(TimeUtils.days(1), TimeUtils.days(1))
-  object Week extends RangeType(TimeUtils.days(7), TimeUtils.days(1))
-  object Month extends RangeType(TimeUtils.days(30), TimeUtils.days(1))
-  object Quarter extends RangeType(TimeUtils.days(92), TimeUtils.days(30))
-  object HalfYear extends RangeType(TimeUtils.days(183), TimeUtils.days(30))
-  object Year extends RangeType(TimeUtils.days(365), TimeUtils.days(30))
-  object Custom extends RangeType(0, 0)
+  object Day      extends RangeType(1.day.toMillis,     1.day.toMillis)
```

Comment:
I think RangeType could also use `Duration` rather than millis? i'm not sure as you'd be carrying nanos but maybe it is a good idea

Reply:
Agree, and it seems like this needs to be parameterized by year to deal with leap years. Maybe investigate as a follow-up because this PR is pretty big already.

Reply:
I created a separate JIRA task for this:
http://jira.gemini.edu:8080/browse/REL-3319